modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -93,76 +101,53 @@ def apply(self, func, *args, **kwargs):
         -----
         It does not matter if `func` is callable or an ``ray.ObjectRef``. Ray will
         handle it correctly either way. The keyword arguments are sent as a dictionary.
+
+        If ``LazyExecution`` is enabled, the function is not applied immediately,
+        but is added to the execution tree.
         """
-        log = get_logger()
-        self._is_debug(log) and log.debug(f"ENTER::Partition.apply::{self._identity}")
-        data = self._data
-        call_queue = self.call_queue + [[func, args, kwargs]]
-        if len(call_queue) > 1:
-            self._is_debug(log) and log.debug(
-                f"SUBMIT::_apply_list_of_funcs::{self._identity}"
-            )
-            result, length, width, ip = _apply_list_of_funcs.remote(
-                data, *deconstruct_call_queue(call_queue)
+        if LazyExecution.get():
+            de = DeferredExecution(self._data_ref, func, args, kwargs)
+            return self.__constructor__(de)
+
+        data = self._data_ref
+        if not isinstance(data, DeferredExecution):
+            flat_args = not has_list_or_de(args)
+            flat_kwargs = not has_list_or_de(kwargs.values())
+            if flat_args and flat_kwargs:
+                result = remote_exec_func.remote(func, data, *args, **kwargs)
+                return self.__constructor__(*result)
+            de = DeferredExecution(
+                data, func, args, kwargs, flat_args, flat_kwargs, num_returns=1
```

Comment:
What do you mean? We can pass to the constructor either DeferredExecution or object ref.

Reply:
I would like to minimize the impact of `DeferredExecution` entity on non-lazy mode, especially where this can be done easily.

The ideal situation:
```python
if LazyExecution.get():
    de = DeferredExecution(self._data_ref, func, args, kwargs)
    return self.__constructor__(de)
else:
   # doesn't use `DeferredExecution` for `self.__constructor__`
```

Reply:
Never mind, it will not work until we called `part.drain_call_queue()` in that function.